packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -335,11 +335,12 @@ const findIntersectionAndUpdateTxStore = ({
               localTx.blockHeader = newTx.blockHeader;
             }
           }
-          // Skip overlapping transactions to avoid duplicates
+          // Skip overlapping transactions to avoid duplicates.
+          // Limit # of stored transactions to last `historicalTransactionsFetchLimit`
           localTransactions = deduplicateSortedArray(
             [...localTransactions, ...newTransactions.slice(localTxsFromSameBlock.length)],
             txEquals
-          );
+          ).slice(-1 * historicalTransactionsFetchLimit);
```

Comment:
@mkazlauskas . I did not add extra changes from your initial impl here. Please let me know if that is ok

Reply:
Not sure if this change has test coverage, but I'm ok merging this as is due to time pressure